presto/dataset.py

```diff
@@ -287,22 +361,79 @@ def __init__(
 
         super().__init__(dataframe)
         if balance:
-            neg_indices, pos_indices = [], []
-            for loc_idx, (_, row) in enumerate(self.df.iterrows()):
-                target = self.target_function(row.to_dict())
-                if target == 0:
-                    neg_indices.append(loc_idx)
+            if self.task_type == "cropland":
+                neg_indices, pos_indices = [], []
+                for loc_idx, (_, row) in enumerate(self.df.iterrows()):
+                    target = self.target_crop(
+                        row, self.task_type, self.croptype_list, return_hierarchical_labels
+                    )
+                    if target == 0:
+                        neg_indices.append(loc_idx)
+                    else:
+                        pos_indices.append(loc_idx)
+                if len(pos_indices) > len(neg_indices):
+                    self.indices = (
```

Comment:
shouldn't we have a similar option to cap the number of repeats for the cropland task?